janus/src/message.rs

```diff
@@ -557,11 +553,11 @@ impl Decode for HpkeKdfId {
 #[repr(u16)]
 pub enum HpkeAeadId {
```

Comment:
Would it break a lot of code if this was just a newtype `HpkeAeadId(Aead)`?

Reply:
That might be better. TBH the only reason we need to define this or the `HpkeKemId` or `HpkeKdfId` enums in crate `janus` is so we can put `prio::codec::{Encode, Decode}` implementations on them and them send them over the wire. Otherwise your label on a `u16` is just as good as ours. I think maybe we could have `prio::codec` provide something like `impl<T> Encode for T where T: Into<u16>`, and then any enum with `repr(u16)` that we derive `num_enum::IntoPrimitive` on can be used. On the other hand, maybe `Into<u16>` is too widespread for that to be a good idea.

Reply:
I gave this a try and it quickly become more complicated than I think it's worth. The short version is that it makes encoding fallible (since `TryFrom` is fallible) and then you have to plumb errors everywhere and I'm not convinced that's better than these newtypes.